extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/filter/AutoServerFilter.java

```diff
@@ -0,0 +1,87 @@
+package io.quarkus.smallrye.openapi.deployment.filter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.microprofile.openapi.OASFilter;
+import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.eclipse.microprofile.openapi.models.servers.Server;
+
+import io.smallrye.openapi.api.models.servers.ServerImpl;
+
+/**
+ * Automatically add default server if none is provided
+ */
+public class AutoServerFilter implements OASFilter {
+
+    private static final String DESCRIPTION = "Auto generated value";
+    private static final String ZEROS = "0.0.0.0";
+    private static final String LOCALHOST = "localhost";
+    private static final String URL_PATTERN = "%s://%s:%d";
+
+    private String defaultScheme;
+    private String defaultHost;
+    private int defaultPort;
+
+    public AutoServerFilter() {
+
+    }
+
+    public AutoServerFilter(String defaultScheme, String defaultHost, int defaultPort) {
+        this.defaultScheme = defaultScheme;
+        this.defaultHost = defaultHost;
+        this.defaultPort = defaultPort;
+    }
+
+    public String getDefaultScheme() {
+        return defaultScheme;
+    }
+
+    public void setDefaultScheme(String defaultScheme) {
```

Comment:
Do we need setters in this class? Also it would be nice to check if the values are non-null in the constructor

Reply:
Also, do we need getters at all? :)

Reply:
Probably not. Let me remove and null check. Hold on